device.go

```diff
@@ -137,32 +139,40 @@ func (d *device) InstallUpdate(image io.ReadCloser, size int64) error {
 	return err
 }
 
-func (d *device) getInactivePartition() (string, error) {
+func (d *device) getInactivePartition() (string, string, error) {
 	inactivePartition, err := d.GetInactive()
 	if err != nil {
-		return "", errors.New("Error obtaining inactive partition: " + err.Error())
+		return "", "", errors.New("Error obtaining inactive partition: " + err.Error())
 	}
 
 	log.Debugf("Marking inactive partition (%s) as the new boot candidate.", inactivePartition)
 
-	partitionNumber := inactivePartition[len(inactivePartition)-1:]
-	if _, err := strconv.Atoi(partitionNumber); err != nil {
-		return "", errors.New("Invalid inactive partition: " + inactivePartition)
+	partitionSlice := strings.SplitAfter(inactivePartition, "p")
```

Comment:
Repeating what is in https://github.com/mendersoftware/meta-mender/pull/469:

This won't work for `/dev/sda2` type of partitions, which is likely to be common on larger boards, such as Intel. I think you will have to count digits backwards from the end of the string.

Suggestion (untested):

```
partitionNumberDecStr := inactivePartition[len(strings.TrimRight(inactivePartition, "0123456789")):]
```

which gives you the final digits.